system/include/emscripten/bind.h

```diff
@@ -1885,9 +1885,12 @@ void constant(const char* name, const ConstantType& v) {
         static_cast<double>(asGenericValue(BT::toWireType(v))));
 }
 
-// EMSCRIPTEN_BINDINGS simple creates a static constructor function which
-// will get included in the program if the translation unit in which it is
-// define gets linked into the program.
-#define EMSCRIPTEN_BINDINGS(name) __attribute__((constructor)) static void __embind_init_##name(void)
+// EMSCRIPTEN_BINDINGS creates a static struct that will initialize the bindings
+// after any other previously defined dynamic initializations, but before main.
```

Comment:
I think the `..which will get included in the program if..` part is still relevent.  This constructor won't run if the TU is not part of the final program.

Also, perhaps a comment such as "Using a C++ constructor here ensures it occurs after any other C++ constructors in this file, which is not true for `__attribute__((constructor))` (they run before C++ constructors in the same file).